TombEngine/Sound/sound.cpp

```diff
@@ -90,9 +90,12 @@ bool LoadSample(char* pointer, int compSize, int uncompSize, int index)
 	BASS_SampleGetInfo(sample, &info);
 	int finalLength = info.length + 44;	// uncompSize is invalid after 16->32 bit conversion
 
-	if (info.freq != 22050 || info.chans != 1)
+
 	{
-		TENLog("Wrong sample parameters, must be 22050 Hz Mono", LogLevel::Error);
+		if (info.freq != TRLE_QUALITY_SAMPLES && info.freq != HIGH_QUALITY_SAMPLES || info.chans != MONO)
+		{
+			TENLog("Incorrect sample parameters, must be either 22050 Hz or 44100 Hz (Mono)", LogLevel::Error);
+		}
 		return false;
```

Comment:
I think this return false needs to be inside if statement.

Reply:
Moreover, why additional bracket block was introduced without any purpose? This whole code part looks weird to me now.